### PR TITLE
fix defect ospkgs can not parse package groups in centos/rhel 5.x #65

### PR DESCRIPTION
**Pull request by immarvin**
fix problem in parsing the package groups/patterns in pkglist;
add support of package groups/patterns removing

### CODE REVIEW

File: xCAT/postscripts/ospkgs [hunk 1]
```diff
@@ -232,7 +232,7 @@ fi
     fi
 
     array_ospkgdirs=($OSPKGDIR)   
-    IFS=$OIFS
+
     array_empty os_path
     index=0
```
Comment (whowutwut) [inline xCAT/postscripts/ospkgs]: Not your code change, but line 228-237 should not have been indented. Would help to clean up the format problem

File: xCAT/postscripts/ospkgs [hunk 2]
```diff
@@ -305,6 +305,7 @@ fi
             
    done    
 
+    IFS=$OIFS
```
Comment (whowutwut) [inline xCAT/postscripts/ospkgs]: The code from line 228 to 308 should not have been indented at all, could you help clean up this formatting issue?

**Comment by whowutwut** [comment]
Can we describe this problem a little better in the Issues? 
For RHEL/CentOS 5.X, when we are doing pkglist groups, it's not parsing correctly?

**Comment by immarvin** [comment]
appended some detail information on this defect in https://github.com/xcat2/xcat-core/issues/65

**Comment by immarvin** [comment]
commit 5eb333038c0ec97c24a42482077ec398ef06844e

1. remove the 'set -x'
2. added comments on not supporting the group removal in ubuntu
3. added more comments on the ospkg code
4. corrected the indent from line 228 to 308

**Comment by whowutwut** [comment]
I'm ok, no more comments.

**Comment by immarvin** [comment]
ok, is it ok to merge the commit ?

**Comment by coldbloodx** [comment]
Hi xCAT bros:

could you please help us to put this fix to branch "2.9.1-pcm".

Thanks ~  :-)

Xian

**Comment by daniceexi** [comment]
For the requirement to put the change to another branch, I think we can just use 'git cherry-pick' to merge the code to another branch instead of 'pull request' since we have reviewed all the changes.

**Comment by immarvin** [comment]
merged into 2.9.1-pcm branch:
https://github.com/xcat2/xcat-core/pull/82

**Pull request merged**